application/utils/spreadsheet.py

```diff
@@ -237,3 +237,59 @@ def write_spreadsheet(title: str, docs: List[Dict[str, Any]], emails: List[str])
     for email in emails:
         sh.share(email, perm_type="user", role="writer")
     return "https://docs.google.com/spreadsheets/d/%s" % sh.id
+
+
+def generate_staggered_csv(database: db.Node_collection, docs: List[defs.CRE]) -> str:
+    maxOffset = 0
+    related = set()
+
+    def add_offset_cre(
+        cre: defs.CRE, database: db.Node_collection, offset: int, visited_cres: Set
+    ) -> List[Dict[str, str]]:
+        nonlocal maxOffset, related
+        maxOffset = max(maxOffset, offset)
+        rows = []
+
+        rows.append({f"CRE{offset}-ID": cre.id})
+        visited_cres.add(cre.id)
+        dbcre = database.get_CREs(external_id=cre.id)
+        if not dbcre:
+            raise ValueError(f"CRE with id {cre.id} not found in the database")
+        cre = dbcre[0]
+        for link in cre.links:
+            if (
+                link.document.doctype == defs.Credoctypes.CRE
+                and link.document.id not in visited_cres
+            ):
+                if link.ltype == defs.LinkTypes.Contains:
+                    rows.extend(
+                        add_offset_cre(
+                            cre=link.document,
+                            database=database,
+                            offset=offset + 1,
+                            visited_cres=visited_cres,
+                        )
+                    )
+                elif link.ltype == defs.LinkTypes.Related:
+                    related.add(link.document.id)
```

Comment:
Currently this code is doing nothing with related CRES, but we need to have them in the template at some point, for the template to be usable for changing the catalog.  We have two options: add them as id|name list in one extra column, or add them as rows, just like we're adding the children (contains) as rows.  I think the latter is easier to read but more confusing. So I vote for one column for the related CRES. This would require building up a dictionary of the related id|name pairs, and then adding them to rows, together with the CRE in a dictionary of two (or one if there is no related CRES.

Reply:
> Currently this code is doing nothing with related CRES, but we need to have them in the template at some point, for the template to be usable for changing the catalog. 

I agree

>  We have two options: add them as id|name list in one extra column, or add them as rows, just like we're adding the children (contains) as rows

We could have a column named "Related CRE" sure, 
then we can add the CRE listed in this column with a related type link,
During importing tehn we have the hierarchy on the left, then one column with related CREs, then an arbitrary number of standard entries. Do we link each standard to both the normal cre and the related one? This seems  a bit confusing for people. 

Since we want to be capturing more complex relationships in a 2 dimensional matrix I think we should consider a more verbose and less context-based format.

e.g.
* perhaps there should be 2 formats, 1 for mapping standards and one for links between cres
* alternatively perhaps everything should be considered a standard, CREs-included, we forget the hierarchy and just support a link between only 2 standards for each CSV, these 2 standards are CRE and any standard, for this we need to explicitly include the relationship type in a column.

In any case, this is beyond the scope of this particular pull request, let's think about this a bit more, let me know when you are free for a brainstorming session

Reply:
The mapping template specifies with which CRE to link standard entries. The related CREs will not get the standard entries aside. I see your point. I guess it can be an option that is off by default to include the related CREs column in the template. So yes: two formats. and indeed not for this PR.